frontend/src/pages/api/auth/[...nextauth].ts

```diff
@@ -0,0 +1,35 @@
+import NextAuth from "next-auth"
+import SpotifyProvider from "next-auth/providers/spotify"
+
+const scope =
+  "playlist-modify-private playlist-modify-public playlist-read-private"
+
+export const authOptions = {
+  providers: [
+    SpotifyProvider({
+      clientId: process.env.SPOTIFY_CLIENT_ID as string,
+      clientSecret: process.env.SPOTIFY_SECRET as string,
```

Comment:
Could we specify the redirectUri here through the env, so it is not hardcoded to a default value? [Example](https://stackoverflow.com/questions/77054618/spotify-nextauth-js-not-working-invalid-client-invalid-redirect-uri).

Reply:
I have seen the same discussion on the StackOverflow page but it's using another version of NextAuth, currently the `SpotifyProvider` doesn't allow to pass redirect_uri as an argument. 
<img width="331" alt="image" src="https://github.com/user-attachments/assets/43684f20-fea8-4a31-801f-11b7eff05a96">

Reply:
I have seen that it can actually be provided as in params:

```js
      authorization: {
        params: {
          scope,
         redirect_uri: "whatever",
        },
```

I am ok leaving it like this given that we will iterate on the frontend soon. However, if we do so, we need to ensure code and documentation is consistent:

1. Add in the README that we are assuming `http://XXX:3000/api/auth/callback/spotify` is added as redirect uri to the app.
2. Remove references to `redirect_uri` variable in the code (e.g. Makefile, env sample).

Reply:
I tried this approach but it doesn't work as expected, it throws an error related to redirect_uri. I will do these points and we can check later.